vertx-web-validation/src/main/java/io/vertx/ext/web/validation/ValidationHandler.java

```diff
@@ -22,15 +37,40 @@ public interface ValidationHandler extends Handler<RoutingContext> {
   String REQUEST_CONTEXT_KEY = "requestParameters";
 
   /**
-   * @deprecated This method duplicates the behavior of {@link ValidationHandlerBuilder#create(SchemaParser)}.
-   *
-   * @param parser a SchemaParser
+   * @param schemaParser a SchemaParser
    * @return an instance of {@link ValidationHandlerBuilder}.
+   * @deprecated This method duplicates the behavior of
+   * {@link ValidationHandlerBuilder#create(io.vertx.json.schema.SchemaRepository)}.
    */
   @Deprecated
   @GenIgnore(GenIgnore.PERMITTED_TYPE)
-  static ValidationHandlerBuilder builder(SchemaParser parser) {
-    return ValidationHandlerBuilder.create(parser);
+  static ValidationHandlerBuilder builder(SchemaParser schemaParser) {
+    // Not the best way to retrieve Vert. instance I guess. We could also try to get it from SchemaRouter
+    // via schemaParser.getSchemaRouter(), but the vertx field is private.
+    FileSystem fs = Vertx.currentContext().owner().fileSystem();
+
+    if (schemaParser instanceof Draft7SchemaParser) {
+      // Draft7SchemaParser was using Draft7
+      return create(SchemaRepository.create(new JsonSchemaOptions().setDraft(DRAFT7)).preloadMetaSchema(fs));
+    } else if (schemaParser instanceof Draft201909SchemaParser) {
+      // Draft201909SchemaParser was using draft201909
+      return create(SchemaRepository.create(new JsonSchemaOptions().setDraft(DRAFT201909)).preloadMetaSchema(fs));
+    } else if (schemaParser instanceof OpenAPI3SchemaParser) {
+      // OpenAPI3SchemaParser was using Draft7 And Supported only OpenAPI 3.0
+      // The baseUri is irrelevant, as it was also not possible before to pass it
+      SchemaRepository repo = SchemaRepository.create(new JsonSchemaOptions().setDraft(DRAFT7)).preloadMetaSchema(fs);
+      // Load the OpenAPI Spec
+      String ref = "https://spec.openapis.org/oas/3.0/schema/2021-09-28";
+      JsonObject raw = new JsonObject(fs.readFileBlocking(ref.substring("https://".length())));
+      repo.dereference(ref, JsonSchema.of(raw));
+      return create(repo);
+    } else {
+      throw new IllegalArgumentException("Passed SchemaParser is not supported");
+    }
+  }
+
+  static ValidationHandlerBuilder builder(SchemaRepository schemaRepository) {
```

Comment:
Looking at the Javadoc, it seems the intent was to get rid of the `builder` method in this class. Perhaps it's better not to add one with the new API?

Reply:
I removed it.